services/report-storage/src/main/kotlin/ReportStorageService.kt

```diff
@@ -71,6 +71,16 @@ class ReportStorageService(
             fetchReport(runId, report.filename)
         } ?: throw ReportNotFoundException(runId, "<from token>")
     }
+
+    /**
+     * Deletes a report with the given [fileName] for the specified [runId].
+     */
+    suspend fun deleteReport(runId: Long, fileName: String) {
+        val key = generateKey(runId, fileName)
+        if (!reportStorage.containsKey(key)) throw ReportNotFoundException(runId, fileName)
+
+        reportStorage.delete(key)
```

Comment:
`delete` returns a `Boolean` that is *false* if the report did not exist. So you could do
`if (!reportStorage.delete(key)) throw ReportNotFoundException(...)`

Reply:
Good point. I was insecure because the implementation in the library could change over time, but the behaviour is even stated in the comment, so the risk of change is relatively low.